src/modules/worker/worker.service.spec.ts

```diff
@@ -0,0 +1,191 @@
+let callCount = 0;
+
+jest.mock('../../infra/y-redis/api.service.js', () => {
+	return {
+		createApiClient: jest.fn().mockImplementation(() => {
+			return {
+				prototype: jest.fn(),
+				get _destroyed() {
+					if (callCount === 0) {
+						callCount++;
+
+						return false;
+					}
+
+					return true;
+				},
+			};
+		}),
+		Api: jest.fn().mockImplementation(() => {
+			return {
+				prototype: jest.fn(),
+				get _destroyed() {
+					if (callCount === 0) {
+						callCount++;
+
+						return false;
+					}
+
+					return true;
+				},
+			};
+		}),
+	};
+});
+
+import { createMock } from '@golevelup/ts-jest';
+import { Test, TestingModule } from '@nestjs/testing';
+import { Logger } from '../../infra/logger/logger.js';
+import { RedisAdapter } from '../../infra/redis/redis.adapter.js';
+import { RedisService } from '../../infra/redis/redis.service.js';
+import { streamMessageReply, xAutoClaimResponse } from '../../infra/redis/testing/x-auto-claim-response.factory.js';
+import { StorageService } from '../../infra/storage/storage.service.js';
+import { WorkerConfig } from './worker.config.js';
+import { WorkerService } from './worker.service.js';
+
+describe(WorkerService.name, () => {
+	let service: WorkerService;
+	let storageService: StorageService;
+	let redisService: RedisService;
+	let redisAdapter: RedisAdapter;
+
+	beforeAll(async () => {
+		const module: TestingModule = await Test.createTestingModule({
+			providers: [
+				WorkerService,
+				{
+					provide: StorageService,
+					useValue: createMock<StorageService>(),
+				},
+				{
+					provide: RedisService,
+					useValue: createMock<RedisService>(),
+				},
+				{
+					provide: Logger,
+					useValue: createMock<Logger>(),
+				},
+				{
+					provide: WorkerConfig,
+					useValue: {
+						WORKER_TRY_CLAIM_COUNT: 1,
+						WORKER_TASK_DEBOUNCE: 1,
+						WORKER_MIN_MESSAGE_LIFETIME: 1,
+					},
+				},
+			],
+		}).compile();
+
+		service = await module.resolve(WorkerService);
+		storageService = module.get(StorageService);
+		redisService = module.get(RedisService);
+
+		const reclaimedTasks = xAutoClaimResponse.build();
+
+		redisAdapter = createMock<RedisAdapter>({
+			redisPrefix: 'prefix',
+			reclaimTasks: jest.fn().mockResolvedValue(reclaimedTasks),
+			getDeletedDocEntries: jest.fn().mockResolvedValue([]),
+			tryClearTask: jest.fn().mockResolvedValue(0),
+			tryDeduplicateTask: jest.fn().mockResolvedValue(undefined),
+			deleteDeleteDocEntry: jest.fn().mockResolvedValue(undefined),
+		});
+
+		jest.spyOn(redisService, 'createRedisInstance').mockResolvedValue(redisAdapter);
```

Comment:
redisService.createRedisInstance.mockResolvedValue*Once*(redisAdapter) ...gehört für mich aber auch in den jeweiligen test setup rein.